lib/i18n/american-english.js

```diff
@@ -137,39 +137,35 @@ class EnglishLanguagePack extends DefaultLanguagePack {
         if (program.isProgram && program.principal !== null)
             sentence = replaceMeMy(sentence);
 
-        if (!sentence.endsWith(' ?') && !sentence.endsWith(' !') && !sentence.endsWith(' .') && coin(0.5, rng))
-            sentence = sentence.trim() + ' .';
+        if (!sentence.endsWith(' ?') && !sentence.endsWith(' !') && !sentence.endsWith(' .')) {
+            if ((forTarget === 'user' && coin(0.5, rng)) || forTarget === 'agent')
+                sentence = sentence.trim() + ' .';
+        }
         if (forTarget === 'user' && sentence.endsWith(' ?') && coin(0.5, rng))
```

Comment:
What does the second part of this commit mean and where is the change about it? (40eeacd, apply contractions consistently: always for "to be")

Reply:
Contractions are "you are" vs "you're". Previously, we'd generate one or the other randomy. Now:
- for the user, we still do randomly (because we want to train robustly)
- for the agent, we always contract "to be" -> "i'm", "you're", "he's", "it's", "she's"
- for the agent, we never contract "would" or "have" -> "i would", "she has", "we have"